src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -24,10 +24,94 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "gcinfo.h"
 #include "gcinfoencoder.h"
 
+//------------------------------------------------------------------------
+// genIsTableDrivenHWIntrinsic:
+//
+// Arguments:
+//    category - category of a HW intrinsic
+//
+// Return Value:
+//    returns true if this category can be table-driven in CodeGen
+//
+static bool genIsTableDrivenHWIntrinsic(HWIntrinsicCategory category)
+{
+    // TODO - make more categories to the table-driven framework
+    const HWIntrinsicCategory drivableCategory    = HW_Category_SimpleSIMD;
+    const HWIntrinsicCategory nondrivableCategory = HW_Category_Special;
+    return (category & drivableCategory) != 0 && (category & HW_Category_Special) == 0;
+}
+
 void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 {
-    NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
-    InstructionSet isa         = compiler->isaOfHWIntrinsic(intrinsicID);
+    NamedIntrinsic      intrinsicID = node->gtHWIntrinsicId;
+    InstructionSet      isa         = compiler->isaOfHWIntrinsic(intrinsicID);
+    HWIntrinsicCategory category    = compiler->categoryOfHWIntrinsic(intrinsicID);
+    int                 numArgs     = compiler->numArgsOfHWIntrinsic(intrinsicID);
+
+    assert((category & HW_Category_NoCodeGen) == 0);
+
+    if (genIsTableDrivenHWIntrinsic(category))
+    {
+
+        GenTree*  op1        = node->gtGetOp1();
+        GenTree*  op2        = node->gtGetOp2();
+        regNumber targetReg  = node->gtRegNum;
+        var_types targetType = node->TypeGet();
+        var_types baseType   = node->gtSIMDBaseType;
+
+        regNumber op1Reg = REG_NA;
+        regNumber op2Reg = REG_NA;
+        emitter*  emit   = getEmitter();
+
+        assert(numArgs >= 0);
+        instruction ins = compiler->insOfHWIntrinsic(intrinsicID, baseType);
+        assert(ins != INS_invalid);
+        unsigned simdSize = node->gtSIMDSize;
+        assert(simdSize != 0);
+
+        switch (numArgs)
+        {
+            case 1:
+                genConsumeOperands(node);
+                op1Reg = op1->gtRegNum;
+                emit->emitIns_R_R(ins, (emitAttr)simdSize, targetReg, op1Reg);
+                break;
+            case 2:
```

Comment:
I don't think this will be correct long-term. Not all 2 arg nodes are `R_R_RM` (some are `R_R_R`, some are `R_RM_R`, etc).